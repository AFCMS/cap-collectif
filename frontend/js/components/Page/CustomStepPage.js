// @flow
import React from 'react';
import { graphql, useLazyLoadQuery } from 'react-relay';
import { useLocation } from 'react-router-dom';
import { insertCustomCode } from '~/utils/customCode';
import type { CustomStepPageQuery as CustomStepPageQueryType } from '~relay/CustomStepPageQuery.graphql';
import WYSIWYGRender from '../Form/WYSIWYGRender';
import DatesInterval from '../Utils/DatesInterval';

type Props = {| +stepId: string |};

const QUERY = graphql`
  query CustomStepPageQuery($stepId: ID!) {
    customStep: node(id: $stepId) {
      ... on Step {
        id
        body
        title
        timeRange {
          startAt
          endAt
        }
        customCode
      }
    }
  }
`;

/*  
The following is just a react encapsulation of the content that was previously
in the src/Capco/AppBundle/Resources/views/Step/show.html.twig file.
It is mandatory to allow frontend navigation within all steps of a project.
This is neither a rework nor a DS migration. It may come later if/when we decide to
refresh this page
*/
export const CustomStepPage = ({ stepId }: Props) => {
  const { state } = useLocation();
  const data = useLazyLoadQuery<CustomStepPageQueryType>(QUERY, {
    stepId: state?.stepId || stepId,
  });

  React.useEffect(() => {
    insertCustomCode(data?.customStep?.customCode);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [data?.customStep?.id]);

  if (!data) return null;
  const { customStep } = data;
  if (!customStep) return null;

  const { title, body, timeRange } = customStep;

  return (
    <>
      <section className="section--alt">
        <div className="container" style={{ paddingTop: 48 }}>
          <h2 className="h2">{title}</h2>
          {timeRange?.startAt && (
            <div className="mb-30 project__step-dates">
              <i className="cap cap-calendar-2-1 mr-10" />
              <DatesInterval startAt={timeRange.startAt} endAt={timeRange.endAt} fullDay />
            </div>
          )}
          <div className="block">
            <WYSIWYGRender value={body} />
          </div>
        </div>
      </section>
    </>
  );
};

export default CustomStepPage;
