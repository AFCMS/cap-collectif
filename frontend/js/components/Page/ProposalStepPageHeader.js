// @flow
import * as React from 'react';
import { graphql, createFragmentContainer } from 'react-relay';
import { FormattedMessage, useIntl } from 'react-intl';
import { useDisclosure } from '@liinkiing/react-hooks';
import {
  isInterpellationContextFromStep,
  isEstablishmentFormStep,
  getProposalLabelByType,
} from '~/utils/interpellationLabelHelper';
import type { ProposalStepPageHeader_step } from '~relay/ProposalStepPageHeader_step.graphql';
import ProposalCreateModal from '../Proposal/Create/ProposalCreateModal';
import Button from '~ds/Button/Button';
import LoginOverlay from '~/components/Utils/LoginOverlay';

type Props = {
  step: ProposalStepPageHeader_step,
};

export const ProposalStepPageHeader = ({ step }: Props) => {
  const intl = useIntl();
  const { isOpen, onOpen, onClose } = useDisclosure(false);
  const projectType = step.project && step.project.type ? step.project.type.title : null;
  const queryCount = step.proposals.totalCount;
  const total = step.allProposals.totalCount;
  const { fusionCount } = step.allProposals;
  const isInterpellation = isInterpellationContextFromStep(step);
  const isEstablishment = isEstablishmentFormStep(step);
  const isProposalForm = step.form && step.form.objectType === 'PROPOSAL';
  const titleTradKey =
    step.form?.objectType === 'ESTABLISHMENT'
      ? getProposalLabelByType(projectType, 'add-establishment')
      : step.form?.objectType === 'PROPOSAL'
      ? getProposalLabelByType(projectType, 'add')
      : 'submit-a-question';
  const tradKeyForTotalCount = isInterpellation
    ? 'interpellation.count_with_total'
    : isEstablishment
    ? 'establishment.count_with_total'
    : isProposalForm
    ? 'proposal.count_with_total'
    : 'question-total-count';
  const tradKeyForCount = isInterpellation
    ? 'interpellation.count'
    : isEstablishment
    ? 'establishment.count'
    : isProposalForm
    ? 'proposal.count'
    : 'count-questions';

  return (
    <>
      <ProposalCreateModal
        title={titleTradKey}
        proposalForm={step.form}
        show={isOpen}
        onClose={onClose}
      />
      <div id="proposal-step-page-header">
        <h3 className="h3 d-ib mb-15">
          {total !== queryCount ? (
            <FormattedMessage
              id={tradKeyForTotalCount}
              values={{
                num: queryCount,
                total,
              }}
            />
          ) : (
            <FormattedMessage
              id={tradKeyForCount}
              values={{
                num: total,
              }}
            />
          )}
          {step.form && step.kind === 'collect' && fusionCount > 0 && (
            <span className="font-weight-300 color-dark-gray">
              {' '}
              <FormattedMessage
                id={
                  isInterpellation || isEstablishment
                    ? 'interpellation.count_fusions'
                    : 'proposal.count_fusions'
                }
                values={{
                  num: fusionCount,
                }}
              />
            </span>
          )}
        </h3>
        {step.form && step.kind === 'collect' && (
          <span className="pull-right mb-20 mt-20">
            <LoginOverlay>
              <Button
                disabled={!step.form?.contribuable}
                variant="primary"
                variantColor="primary"
                variantSize="small"
                onClick={onOpen}
                id="add-proposal">
                {intl.formatMessage({ id: titleTradKey })}
              </Button>
            </LoginOverlay>
          </span>
        )}
      </div>
    </>
  );
};

export default createFragmentContainer(ProposalStepPageHeader, {
  step: graphql`
    fragment ProposalStepPageHeader_step on ProposalStep {
      id
      allProposals: proposals(first: 0) {
        totalCount
        fusionCount
      }
      proposals(
        first: $count
        after: $cursor
        orderBy: $orderBy
        term: $term
        district: $district
        theme: $theme
        category: $category
        status: $status
        userType: $userType
      ) @connection(key: "ProposalStepPageHeader_proposals", filters: []) {
        totalCount
        edges {
          node {
            id
          }
        }
      }
      ... on CollectStep {
        kind
        form {
          id
          objectType
          contribuable
          ...ProposalCreateModal_proposalForm
        }
        project {
          type {
            title
          }
        }
        voteThreshold
      }
      ... on SelectionStep {
        kind
        form {
          contribuable
          id
          objectType
          ...ProposalCreateModal_proposalForm
        }

        project {
          type {
            title
          }
        }
        ...interpellationLabelHelper_step @relay(mask: false)
      }
    }
  `,
});
