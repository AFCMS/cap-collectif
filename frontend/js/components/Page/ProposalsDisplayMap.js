// @flow
import * as React from 'react';
import { createFragmentContainer, graphql } from 'react-relay';
import type { GeoJson, MapOptions } from '../Proposal/Map/ProposalLeafletMap';
import ProposalLeafletMap from '../Proposal/Map/ProposalLeafletMap';
import type { ProposalsDisplayMap_step } from '~relay/ProposalsDisplayMap_step.graphql';
import type { MapTokens } from '../../redux/modules/user';

type RelayProps = {|
  +step: ProposalsDisplayMap_step,
|};

type Props = {|
  ...RelayProps,
  +mapTokens: MapTokens,
  +geoJsons?: Array<GeoJson>,
  +defaultMapOptions: MapOptions,
|};

export const ProposalsDisplayMap = ({ step, ...rest }: Props) => {
  return step.proposals && step.proposals.edges ? (
    <ProposalLeafletMap
      proposals={step.proposals.edges.filter(Boolean).map(edge => edge.node)}
      {...rest}
    />
  ) : null;
};

export default createFragmentContainer(ProposalsDisplayMap, {
  step: graphql`
    fragment ProposalsDisplayMap_step on ProposalStep {
      proposals(
        first: $count
        after: $cursor
        orderBy: $orderBy
        term: $term
        district: $district
        theme: $theme
        category: $category
        status: $status
        userType: $userType
      ) @connection(key: "ProposalsDisplayMap_proposals", filters: []) {
        edges {
          node {
            id
            ...ProposalLeafletMap_proposals
            ...ProposalMapPopover_proposal
          }
        }
      }
    }
  `,
});
