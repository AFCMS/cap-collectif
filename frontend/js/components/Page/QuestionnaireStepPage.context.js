// @flow
import * as React from 'react';

export type Context = {|
  preloadReply: (replyId: string, skipPreload?: boolean) => void,
|};

export const QuestionnaireStepPageContext = React.createContext<Context>({
  preloadReply: () => undefined,
});
