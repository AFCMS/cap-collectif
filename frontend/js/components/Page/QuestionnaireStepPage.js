// @flow
import * as React from 'react';
import { QueryRenderer, graphql } from 'react-relay';
import { loadQuery } from 'relay-hooks';
import { connect } from 'react-redux';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import environment, { graphqlError } from '~/createRelayEnvironment';
import { type GlobalState, type Dispatch } from '~/types';
import { type QuestionnaireStepPageQueryResponse } from '~relay/QuestionnaireStepPageQuery.graphql';
import { Loader } from '~/components/Ui/FeedbacksIndicators/Loader';
import QuestionnaireStepTabs from '../Questionnaire/QuestionnaireStepTabs';
import QuestionnaireReplyPage, {
  queryReply,
} from '~/components/Questionnaire/QuestionnaireReplyPage/QuestionnaireReplyPage';
import { baseUrl } from '~/config';
import ScrollToTop from '~/components/Utils/ScrollToTop';
import { QuestionnaireStepPageContext, type Context } from './QuestionnaireStepPage.context';
import CookieMonster from '~/CookieMonster';

export type PropsNotConnected = {|
  +questionnaireId: ?string,
  +isPrivateResult: boolean,
|};

type Props = {|
  ...PropsNotConnected,
  +dispatch: Dispatch,
  +isAuthenticated: boolean,
  +enableResults: boolean,
|};

const preloadQueryReply = (
  isAuthenticated: boolean,
  replyId: string,
  setReplyPrefetch: any,
  skipPreload?: boolean,
) => {
  const dataReply = loadQuery();

  dataReply.next(
    environment,
    queryReply,
    {
      isAuthenticated,
      replyId,
    },
    { fetchPolicy: 'store-or-network', skip: skipPreload },
  );

  if (skipPreload) return dataReply;

  setReplyPrefetch(dataReply);
};

const component = ({
  error,
  props,
  context,
  dataPrefetch,
}: {
  ...ReactRelayReadyState,
  props: ?QuestionnaireStepPageQueryResponse,
  context: Context,
  dataPrefetch: any,
}) => {
  if (error) return graphqlError;

  if (props) {
    const { questionnaire } = props;
    if (questionnaire) {
      return (
        <QuestionnaireStepPageContext.Provider value={context}>
          <Router basename={questionnaire?.step?.url?.replace(baseUrl, '')}>
            <ScrollToTop />

            <Switch>
              <Route exact path="/">
                <QuestionnaireStepTabs questionnaire={questionnaire} query={props} />
              </Route>

              <Route
                exact
                path="/replies/:id"
                component={routeProps => (
                  <QuestionnaireReplyPage
                    questionnaire={questionnaire}
                    dataPrefetch={dataPrefetch}
                    {...routeProps}
                  />
                )}
              />
            </Switch>
          </Router>
        </QuestionnaireStepPageContext.Provider>
      );
    }

    return graphqlError;
  }

  return <Loader />;
};

export const QuestionnaireStepPage = ({
  questionnaireId,
  isAuthenticated,
  enableResults,
  isPrivateResult,
}: Props) => {
  const [replyPrefetch, setReplyPrefetch] = React.useState(null);
  const anonymousRepliesIds = React.useMemo(
    () => (questionnaireId ? CookieMonster.getAnonymousRepliesIds(questionnaireId) : []),
    [questionnaireId],
  );

  const context = React.useMemo(
    () => ({
      preloadReply: (replyId: string, skipPreload?: boolean) =>
        preloadQueryReply(isAuthenticated, replyId, setReplyPrefetch, skipPreload),
      anonymousRepliesIds,
    }),
    [isAuthenticated, anonymousRepliesIds],
  );

  return questionnaireId ? (
    <QueryRenderer
      environment={environment}
      query={graphql`
        query QuestionnaireStepPageQuery(
          $id: ID!
          $isAuthenticated: Boolean!
          $isNotAuthenticated: Boolean!
          $enableResults: Boolean!
          $anonymousRepliesIds: [ID!]!
        ) {
          questionnaire: node(id: $id) {
            ... on Questionnaire {
              step {
                url
              }
            }
            ...QuestionnaireReplyPage_questionnaire @arguments(isAuthenticated: $isAuthenticated)
            ...QuestionnaireStepTabs_questionnaire
              @arguments(isAuthenticated: $isAuthenticated, enableResults: $enableResults)
          }
          ...QuestionnaireStepTabs_query
            @arguments(
              anonymousRepliesIds: $anonymousRepliesIds
              isNotAuthenticated: $isNotAuthenticated
            )
        }
      `}
      variables={{
        id: questionnaireId,
        isAuthenticated,
        isNotAuthenticated: !isAuthenticated,
        enableResults: enableResults && !isPrivateResult,
        anonymousRepliesIds,
      }}
      render={({ error, props, retry }) =>
        component({ error, props, retry, context, dataPrefetch: replyPrefetch })
      }
    />
  ) : null;
};

const mapStateToProps = (state: GlobalState) => ({
  isAuthenticated: state.user.user !== null,
  enableResults: state.default.features.beta__questionnaire_result || false,
});

export default connect<Props, PropsNotConnected, _, _, _, _>(mapStateToProps)(
  QuestionnaireStepPage,
);
