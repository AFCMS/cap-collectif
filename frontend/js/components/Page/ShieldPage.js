// @flow
import * as React from 'react';
import { FormattedMessage } from 'react-intl';
import { isSubmitting, submit } from 'redux-form';
import { connect } from 'react-redux';
import { Button } from 'react-bootstrap';
import type { Dispatch, State } from '../../types';
import LoginButton from '../User/Login/LoginButton';
import LoginBox from '../User/Login/LoginBox';
import RegistrationButton from '../User/Registration/RegistrationButton';
import RegistrationModal from '~/components/User/Registration/RegistrationModal';
import LoginModal from '~/components/User/Login/LoginModal';
import { loginWithOpenID } from '~/redux/modules/default';

type Props = {|
  showRegistration: boolean,
  submitting: boolean,
  loginWithOpenId: boolean,
  byPassAuth: boolean,
  onSubmit: (e: Event) => void,
|};

const getShieldBody = ({
  showRegistration,
  submitting,
  byPassAuth,
  loginWithOpenId,
  onSubmit,
}: Props): React.Node => {
  if (showRegistration && !loginWithOpenId) {
    return (
      <>
        <LoginButton className="btn--connection btn-block" />
        <div className="mt-10" />
        <RegistrationButton className="btn-block" />
      </>
    );
  }
  if (byPassAuth && loginWithOpenId) {
    return (
      <>
        <LoginBox prefix="" />
      </>
    );
  }
  return (
    <form id="login-form" onSubmit={onSubmit}>
      <LoginBox prefix="" />
      <Button
        id="confirm-login"
        type="submit"
        className="mt-10 btn-block btn-success"
        disabled={submitting}
        bsStyle="primary">
        {submitting ? (
          <FormattedMessage id="global.loading" />
        ) : (
          <FormattedMessage id="global.login_me" />
        )}
      </Button>
    </form>
  );
};

export class ShieldPage extends React.Component<Props> {
  render() {
    return (
      <div id="shield-agent" className="bg-white col-md-4 col-md-offset-4 panel panel-default">
        <LoginModal />
        <RegistrationModal />
        <div className="panel-body">{getShieldBody(this.props)}</div>
      </div>
    );
  }
}

const mapStateToProps = (state: State) => ({
  showRegistration: state.default.features.registration,
  byPassAuth: state.default.features.sso_by_pass_auth,
  loginWithOpenId: loginWithOpenID(state.default.ssoList),
  submitting: isSubmitting('login')(state),
});
const mapDispatchToProps = (dispatch: Dispatch) => ({
  onSubmit: (e: Event) => {
    e.preventDefault();
    dispatch(submit('login'));
  },
});
const connector = connect(
  mapStateToProps,
  mapDispatchToProps,
);
export default connector(ShieldPage);
