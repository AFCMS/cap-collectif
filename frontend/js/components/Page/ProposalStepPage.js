// @flow
import * as React from 'react';
import { connect } from 'react-redux';
import { Row } from 'react-bootstrap';
import { QueryRenderer, graphql } from 'react-relay';
import ProposalListFilters from '../Proposal/List/ProposalListFilters';
import UnpublishedProposalListView from '../Proposal/List/UnpublishedProposalListView';
import DraftProposalList from '../Proposal/List/DraftProposalList';
import Loader from '../Ui/FeedbacksIndicators/Loader';
import ProposalStepPageHeader from './ProposalStepPageHeader';
import StepPageHeader from '../Steps/Page/StepPageHeader/StepPageHeader';
import environment, { graphqlError } from '../../createRelayEnvironment';
import ProposalListView, { queryVariables } from '../Proposal/List/ProposalListView';
import type { Dispatch, FeatureToggles, State } from '../../types';
import type {
  ProposalStepPageQueryResponse,
  ProposalStepPageQueryVariables,
} from '~relay/ProposalStepPageQuery.graphql';
import config from '../../config';
import warning from '../../utils/warning';
import LoginModal from '~/components/User/Login/LoginModal';

type OwnProps = {|
  stepId: string,
  count: number,
|};

type StateProps = {|
  filters: Object,
  order: ?string,
  terms: ?string,
  isAuthenticated: boolean,
  isTipsMeeeEnabled: boolean,
  features: FeatureToggles,
  dispatch: Dispatch,
|};

type Props = {|
  ...OwnProps,
  ...StateProps,
|};

type RenderedProps = {|
  ...ProposalStepPageQueryResponse,
  count: number,
  isAuthenticated: boolean,
  features: FeatureToggles,
  isTipsMeeeEnabled: boolean,
|};

const parseGeoJson = (district: { +geojson: string, +id: string }) => {
  const { geojson, id } = district;
  try {
    return JSON.parse(geojson);
  } catch (e) {
    warning(
      false,
      `Using empty geojson for ${id} because we couldn't parse the geojson : ${geojson}`,
    );
    return null;
  }
};

export const ProposalStepPageRendered = (props: RenderedProps) => {
  const { isTipsMeeeEnabled, viewer, isAuthenticated, features, step, count } = props;
  const [displayMode, setDisplayMode] = React.useState(step?.mainView);

  React.useEffect(() => {
    if (!displayMode && step?.mainView) setDisplayMode(step?.mainView);
  }, [displayMode, setDisplayMode, step]);

  if (!step) {
    return graphqlError;
  }
  const { form } = step;
  if (!form) return null;

  let geoJsons = [];
  if (features.display_map && form.districts) {
    geoJsons = form.districts
      .filter(d => d.geojson && d.displayedOnMap)
      .map(d => ({
        // $FlowFixMe geojson is a non-null string, don't worry Flow
        district: parseGeoJson(d),
        style: {
          border: d.border,
          background: d.background,
        },
      }));
  }
  return (
    <div id="ProposalStepPage-rendered">
      <StepPageHeader step={step} />
      {isAuthenticated && step.kind === 'collect' && <DraftProposalList step={step} />}
      {isAuthenticated && (
        // $FlowFixMe
        <UnpublishedProposalListView step={step} viewer={viewer} />
      )}
      <ProposalStepPageHeader step={step} />
      <ProposalListFilters step={step} setDisplayMode={setDisplayMode} displayMode={displayMode} />
      <ProposalListView
        displayMap={features.display_map}
        isTipsMeeeEnabled={isTipsMeeeEnabled}
        geoJsons={geoJsons}
        step={step}
        count={count}
        displayMode={displayMode}
        viewer={viewer || null}
        defaultMapOptions={{
          center: { lat: form.mapCenter?.lat || 48.8586047, lng: form.mapCenter?.lng || 2.3137325 },
          zoom: form.zoomMap || 10,
        }}
      />
      <LoginModal />
    </div>
  );
};

export class ProposalStepPage extends React.Component<Props> {
  initialRenderVars: Object;

  constructor(props: Props) {
    super(props);
    this.initialRenderVars = {
      term: props.terms,
      ...queryVariables(props.filters, props.order),
    };
  }

  render() {
    const { count, stepId, isAuthenticated, features, isTipsMeeeEnabled } = this.props;

    return (
      <div className="proposal__step-page">
        <QueryRenderer
          environment={environment}
          query={graphql`
            query ProposalStepPageQuery(
              $stepId: ID!
              $cursor: String
              $orderBy: ProposalOrder
              $isAuthenticated: Boolean!
              $isTipsMeeeEnabled: Boolean!
              $count: Int
              $term: String
              $district: ID
              $category: ID
              $status: ID
              $theme: ID
              $userType: ID
              $isMapDisplay: Boolean!
            ) {
              viewer @include(if: $isAuthenticated) {
                ...ProposalListView_viewer
                ...UnpublishedProposalListView_viewer
              }
              step: node(id: $stepId) {
                ... on ProposalStep {
                  id
                  defaultSort
                  voteType
                  statuses {
                    id
                    name
                  }
                  form {
                    mapCenter {
                      lat
                      lng
                    }
                    zoomMap
                    districts(order: ALPHABETICAL) @include(if: $isMapDisplay) {
                      displayedOnMap
                      geojson
                      id
                      border {
                        id
                        enabled
                        color
                        opacity
                        size
                      }
                      background {
                        id
                        enabled
                        color
                        opacity
                        size
                      }
                    }
                  }
                  kind
                  ...ProposalsDisplayMap_step
                  ...StepPageHeader_step
                  ...ProposalListFilters_step
                  ...ProposalListView_step @arguments(count: $count)
                  ...UnpublishedProposalListView_step @arguments(isAuthenticated: $isAuthenticated)
                  ...ProposalStepPageHeader_step
                  ... on CollectStep {
                    private
                    mainView
                    ...DraftProposalList_step @arguments(isAuthenticated: $isAuthenticated)
                  }
                  ... on SelectionStep {
                    mainView
                  }
                }
              }
            }
          `}
          variables={
            ({
              stepId,
              isAuthenticated,
              isTipsMeeeEnabled,
              count: config.isMobile ? 10 : count,
              cursor: null,
              ...this.initialRenderVars,
              isMapDisplay: features.display_map,
            }: ProposalStepPageQueryVariables)
          }
          render={({
            error,
            props,
          }: {
            ...ReactRelayReadyState,
            props: ?ProposalStepPageQueryResponse,
          }) => {
            if (error) {
              return graphqlError;
            }

            if (props) {
              return (
                <ProposalStepPageRendered
                  {...props}
                  count={count}
                  isAuthenticated={isAuthenticated}
                  features={features}
                  isTipsMeeeEnabled={isTipsMeeeEnabled}
                />
              );
            }
            return (
              <Row>
                <Loader />
              </Row>
            );
          }}
        />
      </div>
    );
  }
}

const mapStateToProps = (state: State) => ({
  isAuthenticated: state.user.user !== null,
  filters: state.proposal.filters || {},
  terms: state.proposal.terms,
  order: state.proposal.order,
  features: state.default.features,
  isTipsMeeeEnabled: !!state.default.features.unstable__tipsmeee,
});

export default connect<Props, OwnProps, _, _, _, _>(mapStateToProps)(ProposalStepPage);
