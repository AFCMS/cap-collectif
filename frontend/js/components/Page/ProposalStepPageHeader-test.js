// @flow
/* eslint-env jest */
import * as React from 'react';
import { shallow } from 'enzyme';
import { ProposalStepPageHeader } from '~/components/Page/ProposalStepPageHeader';
import { $fragmentRefs, $refType } from '../../mocks';

describe('<ProposalStepPageHeader />', () => {
  const props = {
    step1: {
      $refType,
      id: 'step1',
      kind: 'selection',
      form: {
        id: 'form1',
        $fragmentRefs,
        isProposalForm: true,
      },
      project: {
        type: {
          title: 'project.types.petition',
        },
      },
      allProposals: {
        totalCount: 5,
        fusionCount: 2,
      },
      proposals: {
        totalCount: 5,
        edges: [
          {
            node: {
              id: 'proposal1',
            },
          },
        ],
      },
    },
    step2: {
      $refType,
      id: 'step1',
      kind: 'selection',
      form: {
        id: 'form1',
        $fragmentRefs,
        isProposalForm: true,
      },
      project: {
        type: {
          title: 'project.types.interpellation',
        },
      },
      allProposals: {
        totalCount: 5,
        fusionCount: 2,
      },
      proposals: {
        totalCount: 5,
        edges: [
          {
            node: {
              id: 'proposal1',
            },
          },
        ],
      },
    },
    step3: {
      $refType,
      id: 'step1',
      kind: 'selection',
      form: {
        id: 'form1',
        $fragmentRefs,
        isProposalForm: false,
      },
      project: {
        type: {
          title: 'project.types.petition',
        },
      },
      allProposals: {
        totalCount: 5,
        fusionCount: 2,
      },
      proposals: {
        totalCount: 5,
        edges: [
          {
            node: {
              id: 'proposal1',
            },
          },
        ],
      },
    },
    step4: {
      $refType,
      id: 'step1',
      kind: 'collect',
      form: {
        id: 'form1',
        $fragmentRefs,
        isProposalForm: true,
      },
      project: {
        type: {
          title: 'project.types.petition',
        },
      },
      allProposals: {
        totalCount: 5,
        fusionCount: 2,
      },
      proposals: {
        totalCount: 5,
        edges: [
          {
            node: {
              id: 'proposal1',
            },
          },
        ],
      },
    },
    step5: {
      $refType,
      id: 'step1',
      kind: 'collect',
      form: {
        id: 'form1',
        $fragmentRefs,
        isProposalForm: true,
      },
      project: {
        type: {
          title: 'project.types.interpellation',
        },
      },
      allProposals: {
        totalCount: 5,
        fusionCount: 2,
      },
      proposals: {
        totalCount: 5,
        edges: [
          {
            node: {
              id: 'proposal1',
            },
          },
        ],
      },
    },
  };

  it('renders with proposal context in selection step', () => {
    const wrapper = shallow(<ProposalStepPageHeader step={props.step1} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with interpellation context in selection step', () => {
    const wrapper = shallow(<ProposalStepPageHeader step={props.step2} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with question context in selection step', () => {
    const wrapper = shallow(<ProposalStepPageHeader step={props.step3} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with proposal context in collect step', () => {
    const wrapper = shallow(<ProposalStepPageHeader step={props.step4} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with interpellation context in collect step', () => {
    const wrapper = shallow(<ProposalStepPageHeader step={props.step5} />);
    expect(wrapper).toMatchSnapshot();
  });
});
