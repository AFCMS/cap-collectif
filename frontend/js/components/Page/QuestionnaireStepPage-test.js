// @flow
/* eslint-env jest */
import * as React from 'react';
import { shallow } from 'enzyme';
import { QuestionnaireStepPage } from './QuestionnaireStepPage';

describe('<QuestionnaireStepPage />', () => {
  const defaultProps = {
    questionnaireId: '1',
    isAuthenticated: true,
    isPrivateResult: false,
    enableResults: true,
  };

  it('renders correctly', () => {
    const wrapper = shallow(<QuestionnaireStepPage {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly when not authenticated', () => {
    const props = {
      ...defaultProps,
      isAuthenticated: false,
    };
    const wrapper = shallow(<QuestionnaireStepPage {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly when enableResults is false', () => {
    const props = {
      ...defaultProps,
      enableResults: false,
    };
    const wrapper = shallow(<QuestionnaireStepPage {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly when enableResults is null', () => {
    const props = {
      ...defaultProps,
      enableResults: null,
    };

    // $FlowFixMe
    const wrapper = shallow(<QuestionnaireStepPage {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
