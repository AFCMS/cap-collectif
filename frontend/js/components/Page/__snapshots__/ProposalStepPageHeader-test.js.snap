// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProposalStepPageHeader /> renders with interpellation context in collect step 1`] = `
<Fragment>
  <h3
    className="h3 d-ib mb-15"
  >
    <FormattedMessage
      id="interpellation.count"
      values={
        Object {
          "num": 5,
        }
      }
    />
    <span
      className="font-weight-300 color-dark-gray"
    >
       
      <FormattedMessage
        id="interpellation.count_fusions"
        values={
          Object {
            "num": 2,
          }
        }
      />
    </span>
  </h3>
  <span
    className="pull-right mb-20 mt-20"
  >
    <ForwardRef(Relay(Connect(injectIntl(ProposalCreate))))
      projectType="project.types.interpellation"
      proposalForm={
        Object {
          "$fragmentRefs": null,
          "id": "form1",
          "isProposalForm": true,
        }
      }
    />
  </span>
</Fragment>
`;

exports[`<ProposalStepPageHeader /> renders with interpellation context in selection step 1`] = `
<Fragment>
  <h3
    className="h3 d-ib mb-15"
  >
    <FormattedMessage
      id="interpellation.count"
      values={
        Object {
          "num": 5,
        }
      }
    />
  </h3>
</Fragment>
`;

exports[`<ProposalStepPageHeader /> renders with proposal context in collect step 1`] = `
<Fragment>
  <h3
    className="h3 d-ib mb-15"
  >
    <FormattedMessage
      id="proposal.count"
      values={
        Object {
          "num": 5,
        }
      }
    />
    <span
      className="font-weight-300 color-dark-gray"
    >
       
      <FormattedMessage
        id="proposal.count_fusions"
        values={
          Object {
            "num": 2,
          }
        }
      />
    </span>
  </h3>
  <span
    className="pull-right mb-20 mt-20"
  >
    <ForwardRef(Relay(Connect(injectIntl(ProposalCreate))))
      projectType="project.types.petition"
      proposalForm={
        Object {
          "$fragmentRefs": null,
          "id": "form1",
          "isProposalForm": true,
        }
      }
    />
  </span>
</Fragment>
`;

exports[`<ProposalStepPageHeader /> renders with proposal context in selection step 1`] = `
<Fragment>
  <h3
    className="h3 d-ib mb-15"
  >
    <FormattedMessage
      id="proposal.count"
      values={
        Object {
          "num": 5,
        }
      }
    />
  </h3>
</Fragment>
`;

exports[`<ProposalStepPageHeader /> renders with question context in selection step 1`] = `
<Fragment>
  <h3
    className="h3 d-ib mb-15"
  >
    <FormattedMessage
      id="count-questions"
      values={
        Object {
          "num": 5,
        }
      }
    />
  </h3>
</Fragment>
`;
